### PR TITLE
fix: Using standard OTLP env convention on backend service

### LINKED ISSUE #1817: [👨🏻‍💻 Internal]: Using standard OTLP env convention on backend service
**Describe the solution you'd like**
Using OTEL_EXPORTER_OTLP_ENDPOINT instead of OTLP_ENDPOINT
Using OTEL_SERVICE_NAME instead of SERVICE_NAME
Extracting protocol from the endpoint instead of OTLP_SPAN_EXPORTER
Supporting OTEL_EXPORTER_OTLP_TRACES_ENDPOINT for traces
Supporting OTEL_EXPORTER_OTLP_LOGS_ENDPOINT for logs
Supporting OTEL_EXPORTER_OTLP_METRICS_ENDPOINT for metrics

**Additional context**
OTLP reference https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/

### PR DESCRIPTION
**Pull request by vr-varad**
Closes #1817

## 📑 Description
Using standard OTLP env convention on backend service

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by vr-varad** [comment]
@talboren I have made the req changes could  u please review it whenever u are free?

File: keep/api/observability.py [hunk 1]
```diff
@@ -26,33 +27,49 @@
 def setup(app: FastAPI):
     logger = logging.getLogger(__name__)
     # Configure the OpenTelemetry SDK
-    service_name = os.environ.get("SERVICE_NAME", "keep-api")
-    otlp_collector_endpoint = os.environ.get("OTLP_ENDPOINT", False)
+    service_name = os.environ.get("OTEL_SERVICE_NAME", "keep-api")
+    otlp_collector_endpoint = os.environ.get("OTEL_EXPORTER_OTLP_ENDPOINT", False)
+    otlp_traces_endpoint = os.environ.get("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", None)
+    otlp_logs_endpoint = os.environ.get("OTEL_EXPORTER_OTLP_LOGS_ENDPOINT", None)
+    otlp_metrics_endpoint = os.environ.get("OTEL_EXPORTER_OTLP_METRICS_ENDPOINT", None)
```
Comment (talboren) [inline keep/api/observability.py:34]: @vr-varad for the sake of backwards compatibility (assuming that some already use that), can we have default values first taken from the old env variables? see example in the next comment

File: keep/api/observability.py [hunk 2]
```diff
@@ -26,33 +27,49 @@
 def setup(app: FastAPI):
     logger = logging.getLogger(__name__)
     # Configure the OpenTelemetry SDK
-    service_name = os.environ.get("SERVICE_NAME", "keep-api")
-    otlp_collector_endpoint = os.environ.get("OTLP_ENDPOINT", False)
+    service_name = os.environ.get("OTEL_SERVICE_NAME", "keep-api")
```
Comment (talboren) [inline keep/api/observability.py:30]:
```suggestion
    service_name = os.environ.get("OTEL_SERVICE_NAME", os.environ.get("SERVICE_NAME", "keep-api"))
```

File: keep/api/observability.py [hunk 3]
```diff
@@ -26,33 +27,49 @@
...
-    else:
-        OTLPSpanExporter = HTTPOTLPSpanExporter
+    def get_protocol_from_endpoint(endpoint):
+        parsed_url = urlparse(endpoint)
+        if parsed_url.scheme == "http":
+            return HTTPOTLPSpanExporter
+        elif parsed_url.scheme == "grpc":
+            return GRPCOTLPSpanExporter
+        else:
+            raise ValueError(f"Unsupported protocol: {parsed_url.scheme}")
```
Comment (talboren) [inline keep/api/observability.py:45]: move function outside

**Review by talboren — CHANGES_REQUESTED** [review]
2 small comments and we're ready to go!

**Comment by talboren** [comment]
@ErezLevip FYI

**Comment by vr-varad** [comment]
@talboren Done!!

**Review by talboren — APPROVED** [review]
LGTM, just need you to sign the CLA :)

**Comment by vr-varad** [comment]
Done @talboren

**Pull request merged**